samples/DotNet/Microsoft.Azure.ServiceBus/SessionSendReceiveUsingSessionClient/Program.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace SessionSendReceiveUsingSessionClient
+{
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.ServiceBus.Core;
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    class Program
+    {
+        const string ServiceBusConnectionString = "{Service Bus connection string}";
+        const string QueueName = "{Queue Name of a Queue that supports sessions}";
+        static IMessageSender messageSender;
+        static ISessionClient sessionClient;
+
+        static void Main(string[] args)
+        {
+            MainAsync().GetAwaiter().GetResult();
+        }
+
+        static async Task MainAsync()
+        {
+            const int numberOfSessions = 5;
+            const int numberOfMessagesPerSession = 3;
+
+            messageSender = new MessageSender(ServiceBusConnectionString, QueueName);
+            sessionClient = new SessionClient(ServiceBusConnectionString, QueueName);
+
+            // Send messages with sessionId set
+            await SendSessionMessagesAsync(numberOfSessions, numberOfMessagesPerSession);
+
+            // Receive all Session based messages using SessionClient
+            await ReceiveSessionMessagesAsync(numberOfSessions, numberOfMessagesPerSession);
+
+            Console.WriteLine("=========================================================");
+            Console.WriteLine("Completed Receiving all messages... Press any key to exit");
+            Console.WriteLine("=========================================================");
+
+            Console.ReadKey();
+
+            await messageSender.CloseAsync();
+            await sessionClient.CloseAsync();
+        }
+
+        static async Task ReceiveSessionMessagesAsync(int numberOfSessions, int messagesPerSession)
+        {
+            while(numberOfSessions-- > 0)
+            {
+                int messagesReceivedPerSession = 0;
+                IMessageSession session = await sessionClient.AcceptMessageSessionAsync();
+                if(session != null)
+                {
```

Comment:
Print sessionId here outside the inner loop. Might make it easier to understand how it works